### PR TITLE
Facebook and cookies

### PR DESCRIPTION
**Pull request by emilyb7**
facebook auth with bell - related #187
save FB user id in cookie - related #188
button on login page links to login route - related #196

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -15,22 +20,32 @@ server.connection({
...
-        directory: {
-          path: path.join(__dirname, '../public'),
-          index: true,
-          defaultExtension: 'html',
-        },
-      },
-    },
-  ]);
+const cookieOptions = {
+  password: 'thi£siVVs&Rgro1ww$MjmakingthisLON7pkassword',
```
Comment (des-des) [inline src/server.js]: dont commit this

Comment (des-des) [inline src/server.js]: make a new one

File: src/server.js [hunk 2]
```diff
@@ -15,22 +20,32 @@ server.connection({
...
+  cookie: 'grow-cookie',
+  isSecure: process.env.NODE_ENV === 'PRODUCTION',
```
Comment (des-des) [inline src/server.js]: wikid

File: src/server.js (hunk 1 above)
Comment (des-des) [inline src/server.js]: derp

File: src/server.js [hunk 3]
```diff
@@ -15,22 +20,32 @@ server.connection({
...
+server.register([ Bell, ], (err) => {
+  if (err) { throw new Error (err); }
+
+  server.auth.strategy('facebook', 'bell', fbOptions);
+});
+
+server.register([ Inert, cookieAuth,], (err) => {
+  if (err) { throw new Error (err); }
+
+  server.auth.strategy('session', 'cookie', 'optional', cookieOptions);
```
Comment (des-des) [inline src/server.js]: `required` rather than `optional`?

Comment (des-des) [inline src/server.js]: Then you can override this on endpoints you do not want auth on (ie login page, or fb callback)

File: src/routes/main.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module.exports = [
+  {
+    path: '/',
+    method: 'GET',
+    handler: (request, reply) => {
+      if (!request.auth.isAuthenticated) {
```
Comment (des-des) [inline src/routes/main.js]: As you have now set the default auth mode to required (see below), you need to explicitly set it to `try` here

File: src/server.js [hunk 5]
```diff
@@ -1,9 +1,14 @@
 const Hapi = require('hapi');
 const Inert = require('inert');
 const fs = require('fs');
-const path = require('path');
-const socket = require('./sockets.js');
+const Request = require('request');
```
Comment (des-des) [inline src/server.js]: Hapi style guide suggests all module variable names start with uppercase, crockford suggests only constructors (used with `new`) should be capitalised. Choose which you are following!

File: src/server.js (hunk 3 above)
Comment (emilyb7) [inline src/server.js]: this breaks everything

Comment (emilyb7) [inline src/server.js]: fixed

**Pull request merged**